assemblers/org.osbuild.qemu

```diff
@@ -455,6 +510,9 @@ def install_grub2(image: str, pt: PartitionTable, options):
             if platform == "powerpc-ieee1275":
                 # write the core to the PrEP partition
                 grub2_write_core_prep_part(core_f, image_f, pt)
+            elif pt.label == "gpt":
+                # gpt requires a bios-boot partition
+                grub2_write_core_bios_boot(core_f, image_f, pt)
```

Comment:
This function also writes to grub's `boot` image on the MBR, which we only write on `i386-pc`? It probably doesn't matter in practice, because the space occupied by the MBR isn't used for anything on gpt, but it did confuse me a bit.

Do you mind splitting that part out of this function to make it bit more apparent what's going on? Or change the name of the function and its docs?

Reply:
Yeah, it "patches" the previously written boot image. I was in two minds about whether it should be in a different function or not so I am more than happy to oblige here ;)

Reply:
This is done. A somewhat interesting side effect is that the core image gets written before the boot image but I guess that is not a big deal.